synapse/push/__init__.py

```diff
@@ -101,7 +101,7 @@
 class PusherConfig:
     """Parameters necessary to configure a pusher."""
 
-    id: Optional[str]
+    id: Optional[int]
```

Comment:
This has had the wrong type forever, but was consistent with itself. Now that we're pulling the proper type from the database layer we had to update it here too.